NewHaven/src/VGMap.cpp

```diff
@@ -0,0 +1,261 @@
+//
+// Created by Nizz Wizz on 2020-02-05.
+//
+
+#include "VGMap.h"
+#include "boost/graph/adjacency_list.hpp"
+#include "boost/graph/graph_utility.hpp"
+#include "boost/graph/connected_components.hpp"
+// include the io stream library
+#include <iostream>
+// include the string library
+#include <string>
+#include <deque>
+using namespace std;
+using namespace boost;
+
+
+VGMap::VGMap() {
+    GenerateGraph();
+}
+// Define the deconstructor of the GameBoard Map
+VGMap::~VGMap() = default;
+// Function that goes and fetches the graph
+// generate the graph
+/*
+The gameboard will be initiated in parts, based on the configuration of the board (aka # of players)
+Each vertex (Square) will be labeled with a number. That number will be indentical to its index in the graph
+and will start at 0 at the upper left most corner of the 2 player field. From there, the vertices will be created row by row.
+It will be created in a similar way as a 2D Matrix.
+*/
+void VGMap::GenerateGraph() {
+    CreateVillageField();
+}
```

Comment:
Why not call CreateVillageField() directly in the constructor instead of calling GenerateGraph()?

Reply:
done